### PR TITLE
change java.util.Random to java.security.SecureRandom

### PR DESCRIPTION
**Pull request by mr-africa**
## :scroll: Description

change `java.util.Random` to `java.security.SecureRandom` for possible security reasons

## :bulb: Motivation and Context

#1830 

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] No breaking changes

### CODE REVIEW

**Review by bruno-garcia — APPROVED** [review]
It's available on [Android since API 1](https://developer.android.com/reference/java/security/SecureRandom) so from that perspective it would be fine.

@maciejwalkowiak @romtsn @marandaneto anything we should be aware before merging?

**Comment by maciejwalkowiak** [comment]
Nothing to be scared of as far as I am aware in regards to running Sentry SDK on server 👍

**Comment by marandaneto** [comment]
@mr-africa you dont need to add a new test, since there's already tests in place for that.

please run `make format`, add an entry to the https://github.com/getsentry/sentry-java/blob/main/CHANGELOG.md and commit :) thanks for doing this

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## Unreleased
 
 * Ref: Rename Fragment span operation from `ui.fragment.load` to `ui.load` (#1824)
+* Ref: change `java.util.Random` to `java.security.SecureRandom` for possible security reasons (#1830)
```
Comment (marandaneto) [inline CHANGELOG.md:6]:
```suggestion
* Ref: change `java.util.Random` to `java.security.SecureRandom` for possible security reasons (#1831)
```

**Comment by marandaneto** [comment]
@mr-africa test should work but you've to fix the import for the Random class in tests, See `TracesSamplerTest`

**Review by marandaneto — APPROVED** [review]

**Pull request merged**